scenario/uploaddata.go

```diff
@@ -149,63 +184,11 @@ func (settings UploadDataSettings) Execute(
 	sessionState.Rest.GetAsync(
 		fmt.Sprintf("%s/%s/quota", host, datafileEndpoint), actionState, sessionState.LogEntry, nil,
 	)
-	sessionState.Rest.GetAsync(
-		fmt.Sprintf(
-			"%s/%s?connectionId=%s&top=1000", host, datafileEndpoint, sessionState.DataConnectionId,
-		), actionState, sessionState.LogEntry, nil,
-	)
-	if sessionState.Wait(actionState) {
-		return // we had an error
+
+	if _, err := sessionState.FetchQixDataFiles(actionState, host, dataConnectionID); err != nil {
+		actionState.AddErrors(errors.WithStack(err))
+		// no return here, wait for async quota too
```

Comment:
Why do we do fetch data files on last time?

Could we wait for some event which tells us upload is finished and data ready to use?

Reply:
for api compliance

Reply:
when upload is finished it's ready to use, so it's not related to that.

Reply:
Alright 👍